zeppelin-web/src/app/credential/credential.controller.js

```diff
@@ -33,24 +48,118 @@ angular.module('zeppelinWebApp').controller('CredentialCtrl', function($scope, $
     }
 
     $http.put(baseUrlSrv.getRestApiBase() + '/credential',
-      { 'entity': $scope.credentialEntity,
-        'username': $scope.credentialUsername,
-        'password': $scope.credentialPassword
-      } ).
+    {
+      'entity': $scope.entity,
+      'username': $scope.username,
+      'password': $scope.password
+    }).
     success(function (data, status, headers, config) {
       BootstrapDialog.alert({
         closable: true,
         message: 'Successfully saved credentials.'
       });
-      $scope.credentialEntity = '';
-      $scope.credentialUsername = '';
-      $scope.credentialPassword = '';
+      getCredentialInfo();
+      $scope.resetCredentialInfo();
+      $scope.showAddNewCredentialInfo = false;
       console.log('Success %o %o', status, data.message);
     }).
     error(function (data, status, headers, config) {
-      alert('Error saving credentials');
+      BootstrapDialog.alert({
+        closable: true,
+        message: 'Error saving credentials'
+      });
       console.log('Error %o %o', status, data.message);
     });
   };
 
+  $scope.cancelCredentialInfo = function() {
+    $scope.showAddNewCredentialInfo = false;
+    $scope.resetCredentialInfo();
+  };
+
+  $scope.resetCredentialInfo = function() {
+    $scope.entity = '';
+    $scope.username = '';
+    $scope.password = '';
+  };
+
+  $scope.copyOriginCredentialsInfo = function(entity) {
```

Comment:
Once the part about credentialInfoTmp is removed, this function won't need any param